### PR TITLE
Add Hygiea, Iapetus, Mercury, Titania, Venus and Leavitt colony tiles

### PR DESCRIPTION
**Pull request by nwai90**
Screenshot 2020-09-29 at 11 26 33 AM

Modified from: https://boardgamegeek.com/thread/2086154/high-orbit-rogue-colonies-8-new-fan-colony-tilesve/page/6

**Notes:**
1) `player.cardDiscount` is introduced: Set to 1 by Iapetus trade and reset to 0 at generation end
2) Building a colony on Mercury allows player to gain the trade income of any active colony in play, without triggering `beforeTrade` or `afterTrade` effects.
- Colony marker does not move, trade fleet is not spent and colony bonuses are not awarded
- `usesTradeFleet` property is introduced for this purpose; default value is `true` to ensure all existing colony behaviour is unchanged
3) `player.colonyVictoryPoints` is introduced to track Titania VP as its own score item
4) `Venus` colony is activated only when a Venus card that collects resources enters play. Reds ruling bonus is applied to Venus raise on colony placement.
5) `player.scienceTagCount` is introduced for Leavitt

### CODE REVIEW

File: src/Game.ts [hunk 1]
```diff
@@ -914,6 +917,7 @@ export class Game implements ILoadable<SerializedGame, Game> {
       this.passedPlayers.clear();
       this.someoneHasRemovedOtherPlayersPlants = false;
       this.players.forEach((player) => {
+        player.cardDiscount = 0;
```
Comment (pierrehilbert) [inline src/Game.ts:920]: Add comment to be sure that we will remember why it's here.

**Review by pierrehilbert — APPROVED** [review]
I think comments could be helpful when we introduce new things in other class (like Player.ts) but the colony seems nicely build.
Vincent is the colony expert, he should give you the final validation

**Review by vincentneko — APPROVED** [review]
This is looking good to me

**Comment by vincentneko** [comment]
@nwai90 can you please resolve the conflict so we can merge this one in master ?

**Comment by nwai90** [comment]
@vincentneko Done

File: src/Player.ts [hunk 2]
```diff
@@ -2040,12 +2046,17 @@ export class Player implements ILoadable<SerializedPlayer, Player>{
...
           && colony.colonies.indexOf(this.id) === -1
-          && colony.isActive);      
-          if (openColonies.length > 0) {
-            standardProjects.options.push(
-                this.buildColony(game, openColonies)
-            );
-          }
+          && colony.isActive);
+
+        if (redsAreRuling && !this.canAfford(constants.BUILD_COLONY_COST + constants.REDS_RULING_POLICY_COST)) {
```
Comment (nwai90) [inline src/Player.ts:1930]: Just realised that this logic should be applicable to Europa colony as well. Happened in a real game which allowed the Reds tax to be avoided. Building a colony on Europa to place an ocean should be disallowed if Reds are ruling and the player cannot pay the mandatory 3 MC tax for the TR increase.

Europa is less straightforward due to the additional 2-12 bonus MC that could potentially be gained from placing adjacent to oceans, which we don't currently factor in. A quick and partial fix would be to add a check for Europa here.

I will submit a separate draft PR that attempts to implement part of the proper logic.

Ref:
https://boardgamegeek.com/thread/2352956/question-about-turmoilreds
https://github.com/bafolts/terraforming-mars/issues/1152

Comment (alrusdi) [reply src/Player.ts:1930]: Damn Reds :-(

**Pull request merged**